views/settings/fields/categorized_multiselect.php

```diff
@@ -0,0 +1,198 @@
+<?php
+namespace WP_Rocket\Views\Settings\Fields\CategorizedMultiselect;
+
+use stdClass;/**
+ * Select field template.
+ *
+ * @param array $data {
+ *     Radio buttons  arguments.
+ *
+ *     @type string $id              Field identifier.
+ *     @type string $label           Field label.
+ *     @type string $container_class Field container class.
+ *     @type string $value           Field value.
+ *     @type string $description     Field description.
+ *     @type array $items            Exclusion list.
+ *     @type array  $options {
+ *          Option options.
+ *
+ *          @type string $description Option value.
+ *          @type string $label Option label.
+ *          @type array  $sub_fields fields to show when option is selected.
+ *     }
+ * }
+ *@since 3.10
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+$wp_rocket_items = key_exists( 'items', $data ) ? $data['items'] : new stdClass();
+
+$wp_rocket_scripts = property_exists( $wp_rocket_items, 'scripts' ) ? $wp_rocket_items->scripts : new stdClass();
+
+$wp_rocket_themes = property_exists( $wp_rocket_items, 'themes' ) ? $wp_rocket_items->themes : new stdClass();
+
+$wp_rocket_plugins = property_exists( $wp_rocket_items, 'plugins' ) ? $wp_rocket_items->plugins : new stdClass();
+
+$wp_rocket_textarea = get_rocket_option( esc_attr( $data['id'] ) );
+if ( is_array( $wp_rocket_textarea ) ) {
+	$wp_rocket_textarea = implode( "\n", $wp_rocket_textarea );
+}
+
+$wp_rocket_select            = get_rocket_option( esc_attr( $data['id'] ) . '_selected' );
+$wp_rocket_select_exclusions = get_rocket_option( esc_attr( $data['id'] ) . '_selected_exclusions' );
+$wp_rocket_state             = json_decode( $wp_rocket_select );
```

Comment:
that's a lot of processing in a view template, this could be in the main class and the final data passed directly